From 463a8faba14f98e377671208858bf8c5fb82a37e Mon Sep 17 00:00:00 2001
From: Richard Acayan <mailingradian@gmail.com>
Date: Fri, 18 Aug 2023 18:46:43 -0400
Subject: [PATCH 043/211] hack! ASoC: dt-bindings: qcom: q6dsp: add internal
 mi2s support

The internal MI2S ports are found on devices with the internal sound
card for Snapdragon 660. Add support for them.

Signed-off-by: Richard Acayan <mailingradian@gmail.com>
---
 .../bindings/sound/qcom,q6dsp-lpass-ports.yaml     |  3 +++
 include/dt-bindings/sound/qcom,q6dsp-lpass-ports.h | 14 ++++++++++++++
 2 files changed, 17 insertions(+)

diff --git a/Documentation/devicetree/bindings/sound/qcom,q6dsp-lpass-ports.yaml b/Documentation/devicetree/bindings/sound/qcom,q6dsp-lpass-ports.yaml
index 3b03e2acd67e..bea6f9f1ef70 100644
--- a/Documentation/devicetree/bindings/sound/qcom,q6dsp-lpass-ports.yaml
+++ b/Documentation/devicetree/bindings/sound/qcom,q6dsp-lpass-ports.yaml
@@ -128,6 +128,7 @@ patternProperties:
                 # MI2S DAI ID range PRIMARY_MI2S_RX - QUATERNARY_MI2S_TX and
                 # QUINARY_MI2S_RX - QUINARY_MI2S_TX and
                 # LPI_MI2S_RX_0 - LPI_MI2S_TX_6
+                # INT0_MI2S_RX - INT6_MI2S_TX
                 items:
                   oneOf:
                     - minimum: 16
@@ -136,6 +137,8 @@ patternProperties:
                       maximum: 128
                     - minimum: 137
                       maximum: 152
+                    - minimum: 153
+                      maximum: 166
         then:
           required:
             - qcom,sd-lines
diff --git a/include/dt-bindings/sound/qcom,q6dsp-lpass-ports.h b/include/dt-bindings/sound/qcom,q6dsp-lpass-ports.h
index 2e879a3c09ae..ecdf29510bd7 100644
--- a/include/dt-bindings/sound/qcom,q6dsp-lpass-ports.h
+++ b/include/dt-bindings/sound/qcom,q6dsp-lpass-ports.h
@@ -156,6 +156,20 @@
 #define LPI_MI2S_TX_5		150
 #define LPI_MI2S_RX_6		151
 #define LPI_MI2S_TX_6		152
+#define INT0_MI2S_RX		153
+#define INT0_MI2S_TX		154
+#define INT1_MI2S_RX		155
+#define INT1_MI2S_TX		156
+#define INT2_MI2S_RX		157
+#define INT2_MI2S_TX		158
+#define INT3_MI2S_RX		159
+#define INT3_MI2S_TX		160
+#define INT4_MI2S_RX		161
+#define INT4_MI2S_TX		162
+#define INT5_MI2S_RX		163
+#define INT5_MI2S_TX		164
+#define INT6_MI2S_RX		165
+#define INT6_MI2S_TX		166
 
 #define LPASS_CLK_ID_PRI_MI2S_IBIT	1
 #define LPASS_CLK_ID_PRI_MI2S_EBIT	2
-- 
2.55.0

